pybop/simulation.py

```diff
@@ -0,0 +1,23 @@
+import pybamm
+
+class Simulation(pybamm.simulation.Simulation):
```

Comment:
What are you thinking of doing with pybamm.Simulation? What functionality are you aiming to re-use?

Reply:
also, again I would use composition here rather than inheritance

Reply:
Initially, for pybop to run the pybamm forward model with the user needing to import pybamm. The longer-term aim is to diverge the parameterisation method so that it can be updated with needing to reconstruct the pybamm.Simulation class (unless there is a method to achieve this already?). [This](https://github.com/BradyPlanden/EstOptParticleModels/blob/6085049d323ba2ede34a79c1195349d71ebe881b/experiment/NLOpt-Fitting.py#L39) is how I've been doing it on a side project.

I wonder if inheritance would be best for this class if we plan to overwrite methods?

Reply:
ok I see. If you want to override methods than inheritance is the way to go. However, I would suggest that it might be good to write pybops own simulation class from scratch, since pybamm's simulation class makes a few assumptions that we do not want:
- it assumes a fixed set of parameters, we want a given set of input parameters
- it assumes a flexible experiment driven by conditions. We want a fixed experiment driven by measured data/timepoints

So our "experiment" and "simulation" classes would probably look a bit different.

Reply:
Note: In the linked python code above you contruct the simulation within the forward model, we definitely don't want to do this, there is a large setup cost for the models and solvers that we want to move outside the foward model. Particularly if you are using JAX